http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -505,12 +513,217 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+        F.raiseError[Unit](AppAlreadyExistsInWorkspaceException(workspaceId, appName, c.app.status, ctx.traceId))
+      )
+
+      samResourceId <- F.delay(AppSamResourceId(UUID.randomUUID().toString))
+      // TODO Verify managed identiy is covered by lookupOriginatingUserEmail
+//      originatingUserEmail <- authProvider.lookupOriginatingUserEmail(userInfo)
+//      _ <- authProvider
+//        .notifyResourceCreated(samResourceId, originatingUserEmail, googleProject)
+//        .handleErrorWith { t =>
+//          log.error(ctx.loggingCtx, t)(
+//            s"Failed to notify the AuthProvider for creation of kubernetes app ${googleProject.value} / ${appName.value}"
+//          ) >> F.raiseError[Unit](t)
+//        }
+
+      cloudContext <- (workspaceDesc.azureContext, workspaceDesc.gcpContext) match {
+        case (Some(azureContext), _) => F.pure[CloudContext](CloudContext.Azure(azureContext))
+        case (_, Some(gcpContext))   => F.pure[CloudContext](CloudContext.Gcp(gcpContext))
+        case (None, None) => F.raiseError[CloudContext](CloudContextNotFoundException(workspaceId, ctx.traceId))
+      }
+
+      saveCluster <- F.fromEither(
+        getSavableCluster(userInfo.userEmail, cloudContext, ctx.now, None, None, Some(workspaceId))
+      )
+      // TODO: Update once AzureCloudContext is populated with AKS information
+      saveClusterResult <- KubernetesServiceDbQueries.saveOrGetClusterForApp(saveCluster).transaction
+      _ <-
+        if (saveClusterResult.minimalCluster.status == KubernetesClusterStatus.Error)
+          F.raiseError[Unit](
+            KubernetesAppCreationException(
+              s"You cannot create an app while a cluster${saveClusterResult.minimalCluster.clusterName.value} is in status ${saveClusterResult.minimalCluster.status}",
+              Some(ctx.traceId)
+            )
+          )
+        else F.unit
+
+      runtimeServiceAccountOpt = none[WorkbenchEmail] // TODO: Figure out how to get PET managed identity
+//        serviceAccountProvider.getClusterServiceAccount(userInfo, googleProject)
+      _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done Sam call for getClusterServiceAccount")))
+      petSA <- F.fromEither(
+        runtimeServiceAccountOpt.toRight(new Exception(s"user ${userInfo.userEmail.value} doesn't have a PET SA"))
+      )
+      // TODO: Adapt the following to support Azure disk
+//      diskResultOpt <- req.diskConfig.traverse(diskReq =>
+//        RuntimeServiceInterp.processPersistentDiskRequest(
+//          diskReq,
+//          config.leoKubernetesConfig.diskConfig.defaultZone, // this need to be updated if we support non-default zone for k8s apps
+//          googleProject,
+//          userInfo,
+//          petSA,
+//          appTypeToFormattedByType(req.appType),
+//          authProvider,
+//          config.leoKubernetesConfig.diskConfig
+//        )
+//      )
+
+      saveApp <- F.fromEither(
+        getSavableApp(
```

Comment:
We should confirm `getSavableApp` populates the right helm chart on both clouds. On GCP we should continue to support our existing charts. For Azure we should only support the `Cromwell` appType and use the chart defined here: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/reference.conf#L209

Reply:
Thinking out loud: should we have a new `AppType` enum for Cromwell-on-Azure?

Reply:
I'm not entirely sure how to address this portion.

Do I add another AppType, create CromwellOnAzureAppConfig with the same fields referenced in your link and add a new case under getSavableApp?

Reply:
What is the resulting data in the DB after `getSavableApp` is called, on both clouds? (Maybe this would be a good unit test).

In particular this logic might need to be changed: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala#L687-L691

Currently it handles 3 app types: Galaxy, Cromwell, and Custom. Cromwell-on-Azure is a separate helm chart though, so want there to be some case which results in populating this information from conf: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/reference.conf#L209-L216

I wasn't sure if we want to make `CromwellOnAzure` a separate AppType, or have logic like:
```
appType match {
  case Galaxy   => config.leoKubernetesConfig.galaxyAppConfig
  case Cromwell if cloud == GCP => config.leoKubernetesConfig.cromwellAppConfig
  case Cromwell if cloud == Azure => config.leoKubernetesConfig.coaAppConfig
  case Custom   => config.leoKubernetesConfig.customAppConfig
}
```